presto-main/src/test/java/com/facebook/presto/sql/planner/TestLogicalPlanner.java

```diff
@@ -370,4 +373,17 @@ public void testCorrelatedScalarAggregationRewriteToLeftOuterJoin()
                                                                                 project(ImmutableMap.of("NON_NULL", expression("true")),
                                                                                         node(ValuesNode.class)))))))))));
     }
+
+    @Test
+    public void testPruneCountAggregationOverScalar()
```

Comment:
@kokosing the second query gets optimized to:

```
- Output[_col0] => [expr_4:boolean]
        Cost: {rows: 1, bytes: ?}
        _col0 := expr_4
    - Project[] => [expr_4:boolean]
            Cost: {rows: 1, bytes: ?}
            expr_4 := true
        - Values => []
                Cost: {rows: 1, bytes: ?}
                ()
```
But this rule applies only to aggregations (getPattern() matches AggregationNode). This rule could be modified to apply to this query but I feel like that should be in different rule. What do you think?